saleor/graphql/file_upload/views.py

```diff
@@ -1,58 +1,170 @@
+    def handle_query(self, request: HttpRequest, *args, **kwargs):
+        try:
+            data = self.parse_body(request)
+        except ValueError as e:
+            return JsonResponse(
+                data={
+                    'errors': [self.format_error('Unable to parse query.')]},
+                status=400)
+        if isinstance(data, list):
+            responses = [self.get_response(request, entry) for entry in data]
+            result = [response for response, code in responses]
+            status_code = max((code for response, code in responses),
+                              default=200)
         else:
-            query, variables, operation_name, id = super(
-                FileUploadGraphQLView,
-                FileUploadGraphQLView).get_graphql_params(request, data)
-        return query, variables, operation_name, id
+            result, status_code = self.get_response(request, data)
+        return JsonResponse(
+            data=result,
+            status=status_code)
+
+    def get_response(self, request: HttpRequest, data: dict):
+        query, variables, operation_name = self.get_graphql_params(
+            request, data)
+        execution_result = self.execute_graphql_request(
+            request, query, variables, operation_name)
+        status_code = 200
+        if execution_result:
+            response = {}
+            if execution_result.errors:
+                response['errors'] = [
+                    self.format_error(e) for e in execution_result.errors]
+            if execution_result.invalid:
+                status_code = 400
+            else:
+                response['data'] = execution_result.data
+            result = response
+        else:
+            result = None
+        return result, status_code
+
+    def get_root_value(self, request: HttpRequest):
+        return self.root_value
+
+    def execute_graphql_request(
+            self, request: HttpRequest, query: str, variables: dict,
+            operation_name: str):
+        if not query:
+            return ExecutionResult(
+                errors=[ValueError('Must provide a query string.')],
+                invalid=True)
+        try:
+            document = self.backend.document_from_string(self.schema, query)
+        except ValueError as e:
+            return ExecutionResult(errors=[e], invalid=True)
+        extra_options = {}
+        if self.executor:
+            # We only include it optionally since
+            # executor is not a valid argument in all backends
+            extra_options['executor'] = self.executor
+        try:
+            return document.execute(
+                root=self.get_root_value(request),
+                variables=variables,
+                operation_name=operation_name,
+                context=request,
+                middleware=self.middleware,
+                **extra_options)
+        except Exception as e:
+            return ExecutionResult(errors=[e], invalid=True)
```

Comment:
That needs to be checked, but if there is an exception, you may be able to catch it here and log the stacktrace to stderr, instead of an unhelpful error message, if you can take a look, otherwise it's fine.

c.f.: https://github.com/graphql-python/graphql-core/issues/208